File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// SquareOfSum calculates the square
// of the sum of the first n integers.
func SquareOfSum(n int) int {
	sum := 0
	for i := 1; i <= n; i++ {
		sum += i
	}
	return sum * sum
}

// SumOfSquares calculates the sum
// of the square of the first n integers.
func SumOfSquares(n int) int {
	sum := 0
	for i := 1; i <= n; i++ {
		sum += i * i
	}
	return sum
}

// Difference calculates the difference between
// the SquareOfSum and SumOfSquares.
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: clojure/difference-of-squares/src/difference_of_squares.clj
(ns difference-of-squares)

(defn numbers [n]
  (range 1 (inc n)))

(defn sum [ns]
  (apply + ns))

(defn square [n]
  (* n n))

(defn sum-of-squares [n]
  (->> (numbers n)
       (map square)
       sum))

(defn square-of-sum [n]
  (->> (numbers n)
       sum
       square))

(defn difference [n]
  (- (square-of-sum n)
     (sum-of-squares n)))
